modules/hitlnext/src/backend/api.ts

```diff
@@ -51,12 +54,26 @@ export default async (bp: typeof sdk, state: StateType, repository: Repository)
     next()
   }
 
+  const hasPermissionOrAgentOnline = (type: string, actionType: 'read' | 'write') => async (
+    req: HITLBPRequest,
+    res: Response,
+    next
+  ) => {
+    const hasPermission = await bp.http.hasPermission(req, actionType, 'module.hitlnext.' + type, true)
+
+    if (hasPermission) {
+      return next()
+    } else {
+      return agentOnlineMiddleware(req, res, next)
+    }
+  }
+
   // Catches exceptions and handles those that are expected
   const errorMiddleware = fn => {
     return (req: BPRequest, res: Response, next) => {
       Promise.resolve(fn(req as BPRequest, res, next)).catch(err => {
         if (err instanceof Joi.ValidationError) {
-          next(new UnprocessableEntityError(formatValidationError(err)))
+          throw new UnprocessableEntityError(formatValidationError(err))
```

Comment:
Pretty sure we don't want to be throwing errors in the error handling middleware. Since ExpressJS v4 does not handle promise rejection by default, throwing in a handler has the side effect of raising an `unhandledRejection` at the process level!

Any specific reason why you don't want to pass the error to the next middleware?

Reply:
Please ignore this change. It was my fault that it was there, but not my intention, thanks for pointing that out